Migrate to general data cube definition (#382)
Open-EO/openeo-processes: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## Changes for vector cubes

> - Update the processes based on `raster-cubes` to work with `datacube` instead

Author: ```suggestion
- Update the processes based on `raster-cube` or `vector-cube` to work with `datacube` instead```

> - Renamed `create_raster_cube` to `create_data_cube`

Author: ```suggestion
- Rename `create_raster_cube` to `create_data_cube`
```

- `add_dimension`: Added new dimension type `vector`
- New definition for `aggregate_spatial`:
    - Allows more than 3 input dimensions

>     - Allow to not export statistics by changing the parameter `target_dimenaion`

Author: ```suggestion
    - Allow to not export statistics by changing the parameter `target_dimension`
```

>     - Clarified how the resulting vector cube looks like

Author: ```suggestion
    - Clarify how the resulting vector cube looks like
```
